salt/runners/vault.py

```diff
@@ -63,7 +63,12 @@ def generate_token(minion_id, signature, impersonated_by_master=False):
                     return {'error': response.reason}
                 config['auth']['token'] = response.json()['auth']['client_token']
 
-        url = '{0}/v1/auth/token/create'.format(config['url'])
+        role_name = config.get('role_name', None)
+        base_url = config['url']
+        if role_name is None
+            url = '{0}/v1/auth/token/create'.format(base_url)
+        else
+            url = '{0}/v1/auth/token/create/{1}'.format(base_url, role_name)
```

Comment:
I think this whole block could be simplified:
```
url = '{0}/v1/auth/token/create'.format(config['url'])
role_name = config.get('role_name')
if role_name:
    url = url + '/' + role_name
```
What do you think?

Reply:
seconded

Reply:
Not that it matters too much but do we run a risk of doubling-up on a slash after URL using this method?

Reply:
We control the while url building in this case, the user cannot pass anything in that would cause an extra slash.